### PR TITLE
Wire element time fields

### PR DESCRIPTION
**Pull request by weswalla**

### CODE REVIEW

File: src/chain_actions/create_action.rs [hunk 1]
```diff
@@ -79,11 +79,13 @@ impl CreateAction {
                 create_link(time_path.hash()?, entry_hash.clone(), ())?;
             }
         }
-
+        let time = sys_time()?; // this won't exactly match the timestamp stored in the element details
```
Comment (Connoropolous) [inline src/chain_actions/create_action.rs:82]: that should be ok

File: src/chain_actions/update_action.rs [hunk 2]
```diff
@@ -60,10 +60,19 @@ impl UpdateAction {
                 create_link(time_path.hash()?, entry_address.clone(), ())?;
             }
         }
+        let updated_at = sys_time()?;
+        // get create time from the header_hash
+        let maybe_element = get::<HeaderHash>(header_hash.clone().into(), GetOptions::default())?;
```
Comment (Connoropolous) [inline src/chain_actions/update_action.rs:65]: what's the benefit of the ::<HeaderHash> here?

File: src/retrieval/get_latest_for_entry.rs [hunk 3]
```diff
@@ -31,50 +31,79 @@ impl GetLatestEntry {
...
-                            Header::Update(update) => update.original_header_address.clone(),
-                            Header::Create(_) => element.header_address().clone(),
-                            _ => {
-                                unreachable!("Can't have returned a header for a nonexistent entry")
-                            }
-                        },
-                        element.header().entry_hash().unwrap().to_owned(),
-                    ))),
-                    None => Ok(None),
+fn entry_and_hashes<
```
Comment (Connoropolous) [inline src/retrieval/get_latest_for_entry.rs:84]:
```suggestion
fn original_header_hash_with_entry<
```

File: src/retrieval/get_latest_for_entry.rs [hunk 4]
```diff
@@ -31,50 +31,79 @@ impl GetLatestEntry {
...
+                    },
                     _ => {
                         let mut sortlist = details.updates.to_vec();
                         // unix timestamp should work for sorting
                         sortlist.sort_by_key(|update| update.header().timestamp().as_millis());
                         // sorts in ascending order, so take the last element
                         let last = sortlist.last().unwrap().to_owned();
-                        Some(get_header_hash(last))
+                        let updated_at = last.header().timestamp();
+                        let maybe_entry_and_hashes = entry_and_hashes(get_header_hash(last), get_options)?;
```
Comment (Connoropolous) [inline src/retrieval/get_latest_for_entry.rs:62]:
```suggestion
                        let maybe_entry_and_hashes = original_header_hash_with_entry(get_header_hash(last), get_options)?;
```

File: src/retrieval/get_latest_for_entry.rs [hunk 5]
```diff
@@ -31,50 +31,79 @@ impl GetLatestEntry {
...
         get_options: GetOptions,
-    ) -> ExternResult<OptionEntryAndHash<T>> {
-        // First, make sure we DO have the latest header_hash address
-        let maybe_latest_header_hash = match get_details(entry_hash.clone(), get_options.clone())? {
+    ) -> ExternResult<Option<WireElement<T>>> {
+        match get_details(entry_hash.clone(), get_options.clone())? {
             Some(Details::Entry(details)) => match details.entry_dht_status {
-                metadata::EntryDhtStatus::Live => match details.updates.len() {
+                metadata::EntryDhtStatus::Live => {
+                    let created_at = details.headers.first().unwrap().header().timestamp(); // TODO: better error handling (avoid unwrap)
```
Comment (Connoropolous) [inline src/retrieval/get_latest_for_entry.rs:38]: yeah this should be ok for now... 

maybe assign `details.headers.first().unwrap()` to a variable though and then you can re use it below on line 43

File: src/retrieval/get_latest_for_entry.rs [hunk 6]
```diff
@@ -31,50 +31,79 @@ impl GetLatestEntry {
...
+                    match details.updates.len() {
                     // pass out the header associated with this entry
-                    0 => Some(get_header_hash(details.headers.first().unwrap().to_owned())),
+                    0 => {
+                        let updated_at = created_at.clone();
```
Comment (Connoropolous) [inline src/retrieval/get_latest_for_entry.rs:42]: right

**Review by Connoropolous — CHANGES_REQUESTED** [review]
This is great logic wise, I think if we change the function name then it's ready

File: src/chain_actions/update_action.rs (hunk 2 above)
Comment (weswalla) [reply src/chain_actions/update_action.rs:65]: from what I remember rust analyzer was giving me feedback to specify the generic type

Comment (weswalla) [reply src/chain_actions/update_action.rs]: figured out a better solution

**Pull request merged**